### ISSUE TITLE
How does the delete mutation works if the action commit the id instead of the full hero object ?

### ISSUE THREAD

**Issue by kRapaille** [johnpapa/vue-getting-started#19]
Hi,

As pointed on twitter, at the end of the last chapter, i don't get how the delete mutation works as it seems that we commit the hero's id in the action, instead of the full hero object.

I'm probably missing something but here's how I read it : 

We delete the hero and get is **ID** as response. We, then, commit the **DELETE_HERO** event with the **deletedHeroId** as parameter.
https://github.com/johnpapa/vue-getting-started/blob/cc05449a6f413e4136adc548ae7e00798801eedf/09-managing-state/end/vue-heroes/src/store/index.js#L41-L44

In the mutation, we are expecting to receive the full hero object and filter on the **id** property of this object. But as seen above, I have the feeling the code should here receive the hero's ID as parameter and then do something like **deletedHeroId.id** which shouldn't work.
https://github.com/johnpapa/vue-getting-started/blob/cc05449a6f413e4136adc548ae7e00798801eedf/09-managing-state/end/vue-heroes/src/store/index.js#L30-L32

To be sure, I checked if the dataservice effectively return the ID, and it seems to be correct. This is not a naming issue in the store.js delete action.
https://github.com/johnpapa/vue-getting-started/blob/cc05449a6f413e4136adc548ae7e00798801eedf/09-managing-state/end/vue-heroes/src/shared/data.service.js#L60-L69

**Comment by johnpapa** [comment]
Nice catch. There are a few things at work here.

1 - The action deletes the hero in the data source. Then it passes the hero id to the mutation, which then tries to find that hero to delete it from the store (this fails to find one and moves along). Then the component re-loads the heroes in the store by querying the data source. This is why it "seems" to work and I missed this originally. The net result is the same, but it is misleading, and this is why you were confused on closer inspection of the code. Nice catch!

2 - The component does not need to reload the heroes. I added this code because despite the delete of the hero in the store (which wasnt happening, but should have been) ,there is a chance that another user modifies the database and this reload will get any changes. I added a comment to clarify this in the code as I do think it's a good practice. https://github.com/johnpapa/vue-getting-started/commit/9d1b333c49f6cb51af23939c6c22e13414e98d6a

3 - The fix I implemented is to pass the hero object from the action to the mutation. This is also in this commit https://github.com/johnpapa/vue-getting-started/commit/9d1b333c49f6cb51af23939c6c22e13414e98d6a

Thanks for the raising this!
